### ISSUE TITLE
Node Labeling for `worker_groups` and `worker_groups_launch_template`

### ISSUE THREAD

**Issue by ArunaLakmal** [terraform-aws-modules/terraform-aws-eks#1690]
## Is your request related to a new offering from AWS?
No

## Is your request related to a problem? Please describe.

I'm migrating an existing setup to Terraform based EKS cluster, with the help of these modules. I have a requirement to use the nodeSelector and taints for some special node schedules. 

Is there any way to use the node labeles with `worker_groups` and `worker_groups_launch_template` please note that `nodeGroups` are not an option and I need specifically to use the `worker_groups`

## Describe the solution you'd like.

Anything that can be fit into the Terraform code it self.

## Additional context

I need to add the specific node label and value. changing the label should be updated in my module and pipeline should update t in a launch template, manual label addition cannot be used on the node. New nodes coming from ASG should come with this node label as well.

Appreciate your valuable thoughts. Thanks!

**Comment by daroga0002** [comment]
you can achieve this by adding additional kubelet arguments in worker group definition which will overwrite default:
https://github.com/terraform-aws-modules/terraform-aws-eks/blob/56e93d77de58f311f1d1d7051f40bf77e7b03524/locals.tf#L67

**Comment by ArunaLakmal** [comment]
@daroga0002 Thank you for the reply and appreciate the support. 
Actually,  I used it like this `"--node-labels=role=infra"` and nodes were not able to connect and I  found it was changed in  the 1.16 release. Thanks to your FAQ it was pointed out. 

Later I  added  the labels like this and I was  able to overcome with the issue `"--node-labels=my.domain.io/role=infra"` (just specifying something) so now I can manage pod selectors. I think I can close  it now.
